src/ssd/ssd_titlebar.c

```diff
@@ -25,6 +25,15 @@ ssd_titlebar_create(struct ssd *ssd)
 {
 	struct view *view = ssd->view;
 	struct theme *theme = view->server->theme;
+
+	/* Here the whole theme changing is more preferable */
+	struct theme custom_theme = { 0 };
+	float custom_color[4];
+	if (window_rules_get_custom_border_color(view, custom_color)) {
+		theme_customize_with_border_color(&custom_theme, custom_color);
+		theme = &custom_theme;
+	}
+
```

Comment:
In addition to moving `theme_customize_with_border_color` to this file as a static function, you should probably just make it an omnibus function that returns a `struct theme` that you can use in one shot, *e.g.*:
```c
struct theme theme = get_theme_for_view(view);
```
where the function looks something like
```c
static struct theme
get_theme_for_view(struct view *view)
{
    float custom_color[4];
    if (!window_rules_get_custom_border_color(view, custom_color)) {
        return *view->server->theme;
    }

    struct theme theme = { 0 };
    /* The rest of your original theme_customize_with_border_color() here */
    return theme;
}